client/src/js/directives/bhSubmit.js

```diff
@@ -42,9 +42,9 @@ function bhSubmitDirective() {
         // the response is a promise, toggle the loading state on
         // fulfillment/rejection
         if (isPromise(response)) {
-          response.finally(function () {
-            FormController.$toggleLoading();
-          });
+          response
```

Comment:
Because the error is not being catched anywhere if I try to reject the promise on the unit test I get the following error:

![Screen Shot 2021-01-24 at 14 19 30](https://user-images.githubusercontent.com/4948776/105638008-3cc67b00-5e4f-11eb-8037-08ace12c6b9c.png)

Reply:
Hi @Buzunda, you are right - this code was not very testable as written.  This is a good solution.

Could you wrap these in a closure?  Basically, instead of `.then(FormController.$toggleLoading)`, write `.then(() => FormController.$toggleLoading())`.  This just makes sure the callback is called in isolation.